[Infra] Match runtime packages to runtime major version

martincostello opened the pull request:
https://github.com/open-telemetry/opentelemetry-dotnet/issues/5973#issuecomment-2960315741

## Changes

Refactoring to match runtime packages to the major version of a target framework.

Unfortunately we can't do this complete for `net8.0` without making breaking changes as we have exposed functionality that depends on new APIs added in the `System.Diagnostics.DiagnosticSource` package for `9.0.0`.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3f86f47ac3e067f03dc98620f1c1be22e9d9f9fb/src/OpenTelemetry.Api/Trace/TracerProvider.cs#L80

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3f86f47ac3e067f03dc98620f1c1be22e9d9f9fb/src/OpenTelemetry.Api/Trace/TelemetrySpan.cs#L375

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3f86f47ac3e067f03dc98620f1c1be22e9d9f9fb/src/OpenTelemetry.Api/Trace/ActivityExtensions.cs#L125

As a compromise for `net8.0`, `8.0.x` packages are used for all dependencies _except_ `System.Diagnostics.DiagnosticSource` which uses `9.0.0` to preserve these use cases.

The downside here is that users running `net8.0` will technically be out of support for that package from May 2026, but that's still the case if we do nothing.

We could mitigate this by moving the special case for `net8.0` to depend on the `10.0.0` package in November with a new release for .NET 10 (if that works) or users can self-mitigate by adding an explicit reference to the `10.0.0` version of that package to their own application without upgrading to .NET 10 immediately.

Moving forward to preserve the intention of this PR we would need to accept that new functionality would be gated to upgrading to the matching version of .NET. For example, a new feature added in .NET 10 would require your application to `net10.0` to use it.

_Alternatively_ we could take an approach where we only expose new APIs to older runtimes when the version is itself an LTS version.

In that case, a new API added in .NET 10 would be supported in all previous versions as we know the package exposing it is LTS, but a new API added in .NET 11 would only be available to `net11.0` and later as that is STS.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

paulomorgado commented on Directory.Packages.props:
```diff
@@ -48,15 +46,18 @@
-    <PackageVersion Include="System.Diagnostics.DiagnosticSource" Version="$(LatestRuntimeOutOfBandVer)" />
+    <PackageVersion Include="System.Diagnostics.DiagnosticSource" Version="$(RuntimePackageVersions)" />
+    <!--
+        However, for .NET 8 we need to use the .NET 9 package as the following APIs have been used to expose user-facing functionality:
+        - System.Diagnostics.Activity.ctor(string, string, IEnumerable<KeyValuePair<string, object>>)
+        - System.Diagnostics.Activity.AddException()
+        - System.Diagnostics.Activity.AddLink()
```
Maybe this could be phrased to indicate that we need to (and can) use the latest version.

We, most likely, can use 10.0.x for all targets, like we can use 9.0.x.

For that matter, I'd use a property specific for this package's version.

martincostello replied:
Depends on the resolution to this comment:

> Moving forward to preserve the intention of this PR we would need to accept that new functionality would be gated to upgrading to the matching version of .NET. For example, a new feature added in .NET 10 would require your application to net10.0 to use it.
>
> Alternatively we could take an approach where we only expose new APIs to older runtimes when the version is itself an LTS version.

The PR as-is is working on the assumption new things need newer TFMs/versions, so the MSBuild as-written is correct as it would only have an exception for `net8.0` and code would need to use `#if NET10_0_OR_GREATER`-like constructs as appropriate for future API additions.

eerhardt commented:
> Build / verify-aot-compat / run-verify-aot-compat (ubuntu-22.04, net8.0) (pull_request)Failing after 1m

Try using `8.0.2` of Microsoft.Extensions.Configuration.Binder. The `8.0.0` package has bugs in the source generator that were fixed in patched versions.

pjanotti reviewed (approved):
This will help with conflict versions of `Microsoft.Extensions.*` when using `opentelemetry-dotnet-auto`

martincostello commented on issue-summary.md:
I've added this to the diff for review.

Will be removed before merge and copied into an actual issue.

eerhardt commented:
FYI - we announced https://devblogs.microsoft.com/dotnet/dotnet-sts-releases-supported-for-24-months/. I don't think that changes the direction in this PR. But it is related information.

martincostello committed:
[Infra] Match runtime packages to runtime version

Use .NET 8 packages with `net8.0`, except `System.Diagnostics.DiagnosticSource`.

Resolves #5973.

martincostello committed:
[Infra] Fix W3cTraceContext tests for net8.0

Attempt to resolve `System.IO.FileNotFoundException : Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=9.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.` caused by reference to OpenTelemetry.Instrumentation.AspNetCore 1.12.0.

martincostello commented:
Dependency versions for NuGet packages before and after this change are shown below.

Changes are as expected:

- `OpenTelemetry@net8.0`
  - `Microsoft.Extensions.Diagnostics.Abstractions`: `9.0.0` 🔽 `8.0.0`
  - `Microsoft.Extensions.Logging.Configuration`: `9.0.0` 🔽 `8.0.0`
- `OpenTelemetry.Api.ProviderBuilderExtensions@net8.0`
  - `Microsoft.Extensions.DependencyInjection.Abstractions`: `9.0.0` 🔽 `8.0.0`
- `OpenTelemetry.Exporter.OpenTelemetryProtocol@net8.0`
  - `Microsoft.Extensions.Configuration.Binder` ➕ `8.0.2` (for native AoT fixes)
- `OpenTelemetry.Extensions.Hosting@net8.0`:
  - `Microsoft.Extensions.Hosting.Abstractions`: `9.0.0` 🔽 `8.0.0`

<details>

<summary>d61f9ef7725b31f9516955063d0312a1f17354c4 (main)</summary>

```json
{
  "OpenTelemetry": {
    ".NETFramework4.6.2": {
      "OpenTelemetry.Api.ProviderBuilderExtensions": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.Diagnostics.Abstractions": "9.0.0",
      "Microsoft.Extensions.Logging.Configuration": "9.0.0"
    },
    "net8.0": {
      "OpenTelemetry.Api.ProviderBuilderExtensions": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.Diagnostics.Abstractions": "9.0.0",
      "Microsoft.Extensions.Logging.Configuration": "9.0.0"
    },
    "net9.0": {
      "OpenTelemetry.Api.ProviderBuilderExtensions": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.Diagnostics.Abstractions": "9.0.0",
      "Microsoft.Extensions.Logging.Configuration": "9.0.0"
    },
    ".NETStandard2.0": {
      "OpenTelemetry.Api.ProviderBuilderExtensions": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.Diagnostics.Abstractions": "9.0.0",
      "Microsoft.Extensions.Logging.Configuration": "9.0.0"
    },
    ".NETStandard2.1": {
      "OpenTelemetry.Api.ProviderBuilderExtensions": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.Diagnostics.Abstractions": "9.0.0",
      "Microsoft.Extensions.Logging.Configuration": "9.0.0"
    }
  },
  "OpenTelemetry.Api": {
    ".NETFramework4.6.2": {
      "System.Diagnostics.DiagnosticSource": "9.0.0"
    },
    "net8.0": {
      "System.Diagnostics.DiagnosticSource": "9.0.0"
    },
    "net9.0": {
      "System.Diagnostics.DiagnosticSource": "9.0.0"
    },
    ".NETStandard2.0": {
      "System.Diagnostics.DiagnosticSource": "9.0.0"
    }
  },
  "OpenTelemetry.Api.ProviderBuilderExtensions": {
    ".NETFramework4.6.2": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.DependencyInjection.Abstractions": "9.0.0"
    },
    "net8.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.DependencyInjection.Abstractions": "9.0.0"
    },
    "net9.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.DependencyInjection.Abstractions": "9.0.0"
    },
    ".NETStandard2.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.DependencyInjection.Abstractions": "9.0.0"
    }
  },
  "OpenTelemetry.Exporter.Console": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.4",
      "System.Text.Encodings.Web": "4.7.2",
      "System.Text.Json": "4.7.2"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4",
      "System.Text.Json": "8.0.5"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4",
      "System.Text.Encodings.Web": "4.7.2",
      "System.Text.Json": "4.7.2"
    }
  },
  "OpenTelemetry.Exporter.InMemory": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    }
  },
  "OpenTelemetry.Exporter.OpenTelemetryProtocol": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    ".NETStandard2.1": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    }
  },
  "OpenTelemetry.Exporter.Prometheus.AspNetCore": {
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    }
  },
  "OpenTelemetry.Exporter.Prometheus.HttpListener": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    }
  },
  "OpenTelemetry.Exporter.Zipkin": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.4",
      "System.Text.Encodings.Web": "4.7.2",
      "System.Text.Json": "4.7.2"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4",
      "System.Text.Json": "8.0.5"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4",
      "System.Text.Encodings.Web": "4.7.2",
      "System.Text.Json": "4.7.2"
    }
  },
  "OpenTelemetry.Extensions.Hosting": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.Hosting.Abstractions": "9.0.0"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.Hosting.Abstractions": "9.0.0"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.Hosting.Abstractions": "9.0.0"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.4",
      "Microsoft.Extensions.Hosting.Abstractions": "9.0.0"
    }
  },
  "OpenTelemetry.Extensions.Propagators": {
    ".NETFramework4.6.2": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4"
    },
    "net8.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4"
    },
    "net9.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4"
    },
    ".NETStandard2.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4"
    }
  },
  "OpenTelemetry.Shims.OpenTracing": {
    ".NETFramework4.6.2": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4",
      "OpenTracing": "[0.12.1, 0.13.0)"
    },
    "net8.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4",
      "OpenTracing": "[0.12.1, 0.13.0)"
    },
    "net9.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4",
      "OpenTracing": "[0.12.1, 0.13.0)"
    },
    ".NETStandard2.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.4",
      "OpenTracing": "[0.12.1, 0.13.0)"
    }
  }
}
```

</details>

<details>

<summary>874054d047981a8cbc0fcde037b3c4f9029030c5</summary>

```json
{
  "OpenTelemetry": {
    ".NETFramework4.6.2": {
      "OpenTelemetry.Api.ProviderBuilderExtensions": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.Diagnostics.Abstractions": "9.0.0",
      "Microsoft.Extensions.Logging.Configuration": "9.0.0"
    },
    "net8.0": {
      "OpenTelemetry.Api.ProviderBuilderExtensions": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.Diagnostics.Abstractions": "8.0.0",
      "Microsoft.Extensions.Logging.Configuration": "8.0.0"
    },
    "net9.0": {
      "OpenTelemetry.Api.ProviderBuilderExtensions": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.Diagnostics.Abstractions": "9.0.0",
      "Microsoft.Extensions.Logging.Configuration": "9.0.0"
    },
    ".NETStandard2.0": {
      "OpenTelemetry.Api.ProviderBuilderExtensions": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.Diagnostics.Abstractions": "9.0.0",
      "Microsoft.Extensions.Logging.Configuration": "9.0.0"
    },
    ".NETStandard2.1": {
      "OpenTelemetry.Api.ProviderBuilderExtensions": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.Diagnostics.Abstractions": "9.0.0",
      "Microsoft.Extensions.Logging.Configuration": "9.0.0"
    }
  },
  "OpenTelemetry.Api": {
    ".NETFramework4.6.2": {
      "System.Diagnostics.DiagnosticSource": "9.0.0"
    },
    "net8.0": {
      "System.Diagnostics.DiagnosticSource": "9.0.0"
    },
    "net9.0": {
      "System.Diagnostics.DiagnosticSource": "9.0.0"
    },
    ".NETStandard2.0": {
      "System.Diagnostics.DiagnosticSource": "9.0.0"
    }
  },
  "OpenTelemetry.Api.ProviderBuilderExtensions": {
    ".NETFramework4.6.2": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.DependencyInjection.Abstractions": "9.0.0"
    },
    "net8.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.DependencyInjection.Abstractions": "8.0.0"
    },
    "net9.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.DependencyInjection.Abstractions": "9.0.0"
    },
    ".NETStandard2.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.DependencyInjection.Abstractions": "9.0.0"
    }
  },
  "OpenTelemetry.Exporter.Console": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "System.Text.Encodings.Web": "4.7.2",
      "System.Text.Json": "4.7.2"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "System.Text.Json": "8.0.5"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "System.Text.Encodings.Web": "4.7.2",
      "System.Text.Json": "4.7.2"
    }
  },
  "OpenTelemetry.Exporter.InMemory": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    }
  },
  "OpenTelemetry.Exporter.OpenTelemetryProtocol": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.Configuration.Binder": "8.0.2"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    ".NETStandard2.1": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    }
  },
  "OpenTelemetry.Exporter.Prometheus.AspNetCore": {
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    }
  },
  "OpenTelemetry.Exporter.Prometheus.HttpListener": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    }
  },
  "OpenTelemetry.Exporter.Zipkin": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "System.Text.Encodings.Web": "4.7.2",
      "System.Text.Json": "4.7.2"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "System.Text.Json": "8.0.5"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "System.Text.Encodings.Web": "4.7.2",
      "System.Text.Json": "4.7.2"
    }
  },
  "OpenTelemetry.Extensions.Hosting": {
    ".NETFramework4.6.2": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.Hosting.Abstractions": "9.0.0"
    },
    "net8.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.Hosting.Abstractions": "8.0.0"
    },
    "net9.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.Hosting.Abstractions": "9.0.0"
    },
    ".NETStandard2.0": {
      "OpenTelemetry": "1.13.1-alpha.0.5",
      "Microsoft.Extensions.Hosting.Abstractions": "9.0.0"
    }
  },
  "OpenTelemetry.Extensions.Propagators": {
    ".NETFramework4.6.2": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5"
    },
    "net8.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5"
    },
    "net9.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5"
    },
    ".NETStandard2.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5"
    }
  },
  "OpenTelemetry.Shims.OpenTracing": {
    ".NETFramework4.6.2": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5",
      "OpenTracing": "[0.12.1, 0.13.0)"
    },
    "net8.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5",
      "OpenTracing": "[0.12.1, 0.13.0)"
    },
    "net9.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5",
      "OpenTracing": "[0.12.1, 0.13.0)"
    },
    ".NETStandard2.0": {
      "OpenTelemetry.Api": "1.13.1-alpha.0.5",
      "OpenTracing": "[0.12.1, 0.13.0)"
    }
  }
}
```

</details>

Generated with this code snippet:

<details>

<summary>Program.cs</summary>

```csharp
#:property PublishAot=false

var packages = new Dictionary<string, Dictionary<string, Dictionary<string, string>>>();

foreach (var file in Directory.GetFiles(args[0], "*.nupkg", SearchOption.TopDirectoryOnly).Order())
{
    // open the file as a zip file
    using var archive = System.IO.Compression.ZipFile.OpenRead(file);
    // find the .nuspec file
    var nuspecEntry = archive.Entries.FirstOrDefault(e => e.FullName.EndsWith(".nuspec", StringComparison.OrdinalIgnoreCase));
    if (nuspecEntry == null) continue;

    using var nuspecStream = nuspecEntry.Open();
    using var reader = new StreamReader(nuspecStream);
    var nuspecContent = reader.ReadToEnd();
    var nuspecXml = System.Xml.Linq.XDocument.Parse(nuspecContent);
    var ns = nuspecXml.Root.GetDefaultNamespace();
    var metadata = nuspecXml.Root.Element(ns + "metadata");
    var id = metadata.Element(ns + "id").Value;
    var deps = metadata.Element(ns + "dependencies");
    var groups = new Dictionary<string, Dictionary<string, string>>();
    if (deps != null)
    {
        foreach (var group in deps.Elements(ns + "group"))
        {
            var tfm = group.Attribute("targetFramework")?.Value;

            if (!groups.ContainsKey(tfm))
            {
                groups[tfm] = [];
            }

            foreach (var dep in group.Elements(ns + "dependency"))
            {
                var depId = dep.Attribute("id").Value;
                var version = dep.Attribute("version")?.Value ?? "*";
                groups[tfm][depId] = version;
            }
        }
    }

    packages[id] = groups;
}

Console.WriteLine(System.Text.Json.JsonSerializer.Serialize(packages, new System.Text.Json.JsonSerializerOptions { WriteIndented = true }));
```

</details>

martincostello committed:
[Infra] Update CHANGELOGs

Update the CHANGELOGs with the dependency changes, whether direct and transitive.

rajkumar-rangaraj committed:
Merge branch 'main' into gh-5973

rajkumar-rangaraj committed:
Merge branch 'main' into gh-5973

rajkumar-rangaraj committed:
Merge branch 'main' into gh-5973

Kielek committed:
Merge branch 'main' into gh-5973

martincostello committed:
Merge branch 'main' into gh-5973
